src/preact/base-element.js

```diff
@@ -651,7 +667,10 @@ function shouldMutationBeRerendered(Ctor, m) {
     const props = Ctor['props'];
     for (const name in props) {
       const def = props[name];
-      if (m.attributeName == def.attr) {
+      if (
+        (def.attr && m.attributeName == def.attr) ||
```

Comment:
Nit: No need for a double condition here, the `==` will fail correctly.
```suggestion
        m.attributeName == def.attr ||
```